### DISCUSSION TITLE
How to adjust distance between bar and x- or y-axis in `create.barplot()`?

### DISCUSSION THREAD

**Discussion by WuSelina** [uclahs-cds/public-R-BoutrosLab-plotting-general#92] [Q&A] [answered]
Does anyone know how to adjust the distance between, for instance, the first bar in a barplot and the x- or y-axis? The example I have here is a horizontal, stacked barplot: 
<img width="288" alt="example-barplot-help" src="https://user-images.githubusercontent.com/68160979/217912432-febe88e6-62b7-4f8f-b6ef-68c4fea54ffb.png">

Thank you!!

**Comment by stefaneng** [comment, marked as answer]
@WuSelina I think what you want is to set `ylimits`, or `xlimits` if vertical.

For example:

``` r
library(BoutrosLab.plotting.general)

barplot.data <- data.frame(
    samples = c(1L, 1L, 2L, 2L, 3L, 3L, 4L, 4L, 5L, 5L, 6L, 6L, 7L, 7L),
    values = c(8, 2, 2, 1, 27, 2, 6, 2, 4, 3, 2, 0, 4, 2),
    mutation = c("nonsynonymous", "other", "nonsynonymous", 
    "other", "nonsynonymous", "other", "nonsynonymous", "other", 
    "nonsynonymous", "other", "nonsynonymous", "other", "nonsynonymous", 
    "other")
    )

create.barplot(
    formula = samples ~ values,
    data = barplot.data,
    groups = mutation,
    col = default.colours(2),
    plot.horizontal = TRUE,
    stack = TRUE,
    ylimits = c(0.6, 7.4),
    yat = barplot.data$samples
)
```

![](https://i.imgur.com/V4Gc6p1.png)

<sup>Created on 2023-02-14 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by WuSelina** [comment]
If anyone wants to recreate the duplicated reads QC plot, you can access it on the ["Example BPG Plots" page](https://confluence.mednet.ucla.edu/display/BOUTROSLAB/Example+BPG+Plots). Don't forget about this helpful plotting resource too!

**Comment by pboutros** [comment]
Just a note that the examples are internal, whereas this github repo is public.  Can you create an issue so that we eventually port all examples and hall-of-fame plots over to here at some point?

**Reply by dan-knight** [reply]
Good idea. Issue [here](https://github.com/uclahs-cds/public-R-BoutrosLab-plotting-general/issues/100).
